engines/alcachofa/alcachofa.cpp

```diff
@@ -0,0 +1,435 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/scummsys.h"
+#include "common/config-manager.h"
+#include "common/debug-channels.h"
+#include "common/events.h"
+#include "common/savefile.h"
+#include "common/system.h"
+#include "engines/util.h"
+#include "graphics/paletteman.h"
+#include "graphics/framelimiter.h"
+#include "graphics/thumbnail.h"
+#include "image/png.h"
+#include "video/avi_decoder.h"
+#include "video/mpegps_decoder.h"
+
+#include "alcachofa.h"
+#include "metaengine.h"
+#include "console.h"
+#include "detection.h"
+#include "player.h"
+#include "rooms.h"
+#include "script.h"
+#include "global-ui.h"
+#include "menu.h"
+#include "debug.h"
+#include "game.h"
+
+using namespace Math;
+
+namespace Alcachofa {
+
+constexpr uint kDefaultFramerate = 100; // the original target framerate, not critical
+
+AlcachofaEngine *g_engine;
+
+AlcachofaEngine::AlcachofaEngine(OSystem *syst, const ADGameDescription *gameDesc)
+	: Engine(syst)
+	, _gameDescription(gameDesc)
+	, _eventLoopSemaphore("engine") {
+	assert(gameDesc != nullptr);
+	g_engine = this;
+}
+
+AlcachofaEngine::~AlcachofaEngine() {
+}
+
+uint32 AlcachofaEngine::getFeatures() const {
+	return _gameDescription->flags;
+}
+
+Common::String AlcachofaEngine::getGameId() const {
+	return _gameDescription->gameId;
+}
+
+Common::Error AlcachofaEngine::run() {
+	g_system->showMouse(false);
+	setDebugger(_console);
+	_game.reset(Game::createForMovieAdventure());
+	_renderer.reset(IRenderer::createOpenGLRenderer(Common::Point(1024, 768)));
```

Comment:
The game looked 2d for me. Is it truly using 3d features? Forcing it to OpenGL would severely restrict compatibility.